cardano-client-demo/ChainSyncClientWithLedgerState.hs

```diff
@@ -86,21 +86,21 @@ main = do
 -- | Defines the client side of the chain sync protocol.
 chainSyncClient
   :: ChainSyncClient
-        (BlockInMode CardanoMode, Either Text LedgerState)
+        (BlockInMode CardanoMode, Either Text (LedgerState, [LedgerEvent]))
```

Comment:
:+1: 

BTW, remind me what the `Either Text` is about.

Reply:
We might get an error when applying blocks. The error will be either a `HardForkLedgerError '[...]` from calling `tickThenApplyLedgerResult` or an hash miss-match. I think I originally just used `Text` here out of laziness/simplicity but am happy to use a structured error instead (I'd do that in a separate PR).

Reply:
A structured error  would be great :+1: